analysis/analysis.py

```diff
@@ -356,9 +591,81 @@ def setUp(self):
                 'scroll_ups': 0,
             }
         ]
+        test_data_2_path = Path('data', 'test_data_2.csv')
+        self.default_student_data_2 = load_data_csv(test_data_2_path)
         sample_csv_path = Path('data', 'rereading_data_2019-09-13.csv')
         self.student_data = load_data_csv(sample_csv_path)
 
+    def test_mean_rereading_time_for_a_question(self):
+        """
+        Tests mean_rereading_time_for_a_question function with many data sets and checks if
+        the function crashes when it encounters the default data set. Also test many cases with
+        all question and context combinations.
+        """
+        mean_rereading_data = mean_rereading_time_for_a_question(self.default_student_data, "", "")
+
+        empty_comparison_tuple = ("", "", 0, 0)
+        self.assertEqual(mean_rereading_data, empty_comparison_tuple)
+
+        feel = "In one word, how does this text make you feel?"
+        about = "In three words or fewer, what is this text about?"
+        encountered = "Have you encountered this text before?"
+        ads = "This is an ad."
+        short_story = "This is actually a short story."
+
+        mean_rereading_time_results_data = [
```

Comment:
Rather than using an assertion against 6 tuples in an array, this is a circumstance where calling 6 different assertions against each of these results would be better. That way if something breaks, we'll know which specific assertion broke. (right now if any one of these 6 results fails, we'll get an error message that _something_ broke, but we won't know which call!)